### ISSUE TITLE
SDL1 Joystick problems in DOSBox-X

### ISSUE THREAD

**Issue by rderooy** [joncampbell123/dosbox-x#1823]
**Describe the bug**
I'm testing joysticks on Linux and encountering some strange behaviour.

I'm testing with the following utilities
* DOS joystick utilities: https://www.vogons.org/viewtopic.php?p=187168#p187168
* SDL & SDL2 joystick tests: https://gitlab.com/sdl-jstest/sdl-jstest

When I plug in both a PS4 and PS3 controller, and run the SDL1 or SDL2 tests, both controllers are recognised and work.

Both SDL1 and SDL2 test reports the PS3 controller as joystick0 and the PS4 controller as joystick1:
```
$ ./sdl2-jstest --list
Found 2 joystick(s)

Joystick Name:     'Sony PLAYSTATION(R)3 Controller'
Joystick GUID:     030000004c0500006802000011810000
Joystick Number:    0
Number of Axes:     6
Number of Buttons: 17
Number of Hats:     0
Number of Balls:    0
GameControllerConfig:
  Name:    'PS3 Controller'
  Mapping: '030000004c0500006802000011810000,PS3 Controller,a:b0,b:b1,back:b8,dpdown:b14,dpleft:b15,dpright:b16,dpup:b13,guide:b10,leftshoulder:b4,leftstick:b11,lefttrigger:a2,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b12,righttrigger:a5,rightx:a3,righty:a4,start:b9,x:b3,y:b2,platform:Linux,'

Joystick Name:     'Sony Interactive Entertainment Wireless Controller'
Joystick GUID:     030000004c050000cc09000011810000
Joystick Number:    1
Number of Axes:     6
Number of Buttons: 13
Number of Hats:     1
Number of Balls:    0
GameControllerConfig:
  Name:    'PS4 Controller'
  Mapping: '030000004c050000cc09000011810000,PS4 Controller,a:b0,b:b1,back:b8,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,guide:b10,leftshoulder:b4,leftstick:b11,lefttrigger:a2,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b12,righttrigger:a5,rightx:a3,righty:a4,start:b9,x:b3,y:b2,platform:Linux,'
```

When I then start DOSBox-X SDL1, it shows the following in the log:
```
LOG: Two or more joysticks reported, initializing with 2axis
LOG: Using joystick Sony Interactive Entertainment Wireless Controller with 6 axes, 13 buttons and 1 hat(s)
LOG: Using joystick Sony PLAYSTATION(R)3 Controller with 6 axes, 17 buttons and 0 hat(s)
```
Basically the order got reversed from how jstest reported them. This does not happen with the SDL2 DOSBox-X build.

When I then run JOYCHECK.EXE the reversing is confirmed, the PS4 controller is the first controller and the inputs work fine. But there is another issue, the PS3 controller does NOT work at all (again, this issue is limited to the SDL1 DOSBox-X build, the SDL2 build is not effected).

Running DOSBox-X with ONLY the PS3 controller attached shows in the log:
```
LOG: One joystick reported, initializing with 4axis
LOG: Using joystick Sony PLAYSTATION(R)3 Controller with 6 axes, 17 buttons and 0 hat(s)
```
Running JOYCHECK.EXE again, the PS3 controller now works.

**Comment by rderooy** [comment]
Unplugging both controllers, and first plugging in the PS4 and then the PS3 controller and running the jstest shows that the PS4 controller is now joystick0 while the PS3 controller is now joystick1.

Running SDL1 DOSbox-X, it now shows the correct order in the log. And running JOYCHECK.EXE the PS4 controller is now Joystick1 in DOS. But the PS3 controller on Joystick2 still does not work.

**Comment by rderooy** [comment]
More weird behaviour, but now with the **SDL2 DOSBox-X version**. Without unplugging the controllers, I ran DOSBox-X SDL2.

The log shows
```
LOG: Two or more joysticks reported, initializing with 2axis
LOG: Using joystick Sony PLAYSTATION(R)3 Controller with 6 axes, 17 buttons and 0 hat(s)
LOG: Using joystick Sony Interactive Entertainment Wireless Controller with 6 axes, 13 buttons and 1 hat(s)
```

So in this instance the order also got reversed with the SDL2 build, and running JOYCHECK.EXE the PS4 controller as Joystick2 works, but the PS3 controller as Joystick1 does not.

**Comment by rderooy** [comment]
partly false report. The PS3 controller simply needed me to press the PS button to get them working. But the ordering problem is real. WIll close this report to prevent confusion.
